### ISSUE TITLE
Page view goals break when using both foo and foo/**

### ISSUE THREAD

**Issue by NathanC** [plausible/analytics#2399]
### Past Issues Searched

- [X] I have searched open and closed issues to make sure that the bug has not yet been reported

### Issue is a Bug Report

- [X] This is a bug report and not a feature request, nor asking for self-hosted support

### Using official Plausible Cloud hosting or self-hosting?

Self-hosting

### Describe the bug

I have products on my page, and I expose both of these path patterns:

* `example.com/products` -> a gallery displaying products
* `example.com/products/bar` -> a single product page for the product "bar"

I'd like to have seperate goals for viewing the product gallery (`/product`) or viewing an individual product page (`/product/*`). This is especially useful because I can segment custom events based on those goals (to see if they added a product to the cart from the gallery, or from the individual page).

Setting up two custom page view goals (`/product` and `/product/**` or `/product/*`) doesn't seem to work properly. Only one of these goals is tracked, usually. The other just gets dropped.

It's very inconsistent though-- there seems to be some (not quite reproducible differences) between `product/*` and `product/**` when also using `/product`. It seems like the first one of these goals that is met might become sticky, and prevent the other goal from being met.

I'm using the same pattern (1 level path for gallery, 2 level path for individual) for both products and sellers, and I set them up the exact same way (`/products` and `/products/*` + `/sellers` and `/sellers/*`). Weirdly, it worked for sellers, but not for products. By working, I mean it tracked the goals separately.

Is this expected behavior due to the way path prefixes are analyzed? Do I need to change `/products` to `/gallery` or something, with `/products/foo` still being the individual page? I'd really like to avoid that if possible.

### Expected behavior

`products` and `products/*` (or `products/**`) are tracked as separate page view goals.

### Environment

```markdown
- OS: Windows 11
- Browser: Firefox
- Browser Version: 106.0.2

I'm self hosting, but I'm using the latest RC (`1.5.0-rc.0`)
```

**Comment by metmarkosaric** [comment]
hi @NathanC! pageview goals basically just show what's in your Top Pages report. there's no special algorithm or anything like that behind them. what you see in pageviews goals should fit with what you see in Top Pages or what you see when you use the "filter" button on the top of your dashboard. you can try to link to specific examples or try to replicate this on the cloud as everything seems to be working as intended

**Comment by NathanC** [comment]
@metmarkosaric  hmm, in that case I think this is likely an actual bug. My page goals for these specific patterns definitely do not reflect the top pages listed. 

I'm actually using `/artists` and `/artwork` (not `/users` and `/products` like I posted in the bug report-- I was trying to make it more generic). Anyhow, here are some screenshots that demonstrate the issue. I deleted all the site data and then went to a few pages for this test:

# Goals

![image](https://user-images.githubusercontent.com/6182419/198533122-edb4c1b9-de75-4f3c-a0e5-8f2cd111b857.png)

# Top Pages

![image](https://user-images.githubusercontent.com/6182419/198533293-07c9e090-b1e5-449e-a8ae-2fb1a788f49c.png)

# Goal Conversions

![image](https://user-images.githubusercontent.com/6182419/198533365-75dbff44-7197-42ae-ac31-936bddab018d.png)

# What does this mean?

Note how all 4 goals are represented, but only 3 display. The top pages and goal conversions are screenshots from the same dashboard view, without any changes in filters or ranges or anything-- just 2 separate screenshots of the combined view I see.

This behavior is inconsistent though. After I reset the site to get these screenshots for you, both the `/artists` and `/artists/*` goals are being tracked, even though `/artwork/*` isn't being tracked. Before, it wasn't working quite like that, I believe either artwork had 2 goals or neither artists nor artwork had 2 goals.

Do you see the issue I mean, or do I need to clarify more?

**Comment by metmarkosaric** [comment]
the data seems all fine to me and consistent with the top pages report. when adding a new pageview goal, make sure not to have any blank spaces before or after the name as otherwise the goal won't track correctly. i have tested this on our version and i don't see any issues. everything is working as expected

**Comment by NathanC** [comment]
@metmarkosaric You see how the `Visit /artwork` goal is missing from the conversions in the last screenshot I gave, right? That isn't supposed to happen, is it?

I did just delete my page view goal for `/artwork` and recreate it, and now it seems to be capturing all 4 goals-- it's definitely possible that I had a leading or trailing space, though that not being seen in the ui would be a minor bug in itself. I'm not quite convinced though-- might just be hard to reproduce.

Regardless, it's working now, so all's good :) If the issue crops up again I'll add to this thread. Thank you very much for your prompt response, and for creating such a useful product!
